packages/react/src/transactions/useApproveModule.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  InsufficientGasError,
+  PendingSigningRequestError,
+  TransactionKind,
+  UserRejectedError,
+  WalletConnectionError,
+} from '@lens-protocol/domain/entities';
+import { SupportedTransactionRequest } from '@lens-protocol/domain/use-cases/transactions';
+import {
+  TokenAllowanceLimit,
+  TokenAllowanceRequest,
+} from '@lens-protocol/domain/use-cases/wallets';
+import { Amount, Erc20, EthereumAddress } from '@lens-protocol/shared-kernel';
+import { useState } from 'react';
+
+import {
+  TransactionState,
+  useHasPendingTransaction,
+  useHasSettledTransaction,
+} from './adapters/TransactionQueuePresenter';
+import { useApproveModuleController } from './adapters/useApproveModuleController';
+
+export type UseApproveModuleArgs = {
+  amount: Amount<Erc20>;
+  spender: EthereumAddress;
+};
+
+export function useApproveModule({ amount, spender }: UseApproveModuleArgs) {
+  const [isPending, setIsPending] = useState(false);
+  const [error, setError] = useState<
+    | InsufficientGasError
+    | PendingSigningRequestError
+    | UserRejectedError
+    | WalletConnectionError
+    | null
+  >(null);
+  const setAllowance = useApproveModuleController();
+
+  function isApproveTransaction(
+    transaction: TransactionState<SupportedTransactionRequest>,
+  ): transaction is TransactionState<TokenAllowanceRequest> {
+    return (
+      transaction.request.kind === TransactionKind.APPROVE_MODULE &&
+      transaction.request.spender === spender &&
+      transaction.request.amount.asset === amount.asset
+    );
+  }
```

Comment:
I feel like we need to debate this predicate, if you test the example app you'll see what i mean. Let's say i have a UI that shows a 2 publications, both of them with a paid collectable module and the same asset. If the user approves the first one  with an amount that is not enough for the second one, the second one will always appear as "approved" because we're not checking the quantity here. Also, if we check the quantity is the same if the two publications use the same amount. Should we check by `id` ?

Reply:
We could for now assume that in normal circumstances the frontend should wait before trying to approve a new module (maybe we can even throw here an error if trying to approve more than one)
I would personally also remove `isApproved` flag as it may be confusing for the users. It doesn't say whether the given module is approved it just checks the transaction history to see if you made the transaction to approve the given module.